### ISSUE TITLE
Re-binding mouse buttons should have a GUI exception (potentially optional)

### ISSUE THREAD

**Issue by meganukebmp** [anegostudios/VintageStory-Issues#4041]
### Game Version

v1.19.8

### Platform

Windows

### Modded

None

### SP/MP

None

### Description

When rebinding the mouse buttons (for example right click to break and left click to place), this also rebinds the buttons within GUIs (inventories for example) which isnt necessarily what's wanted. Other voxel games usually allow for changing the buttons without affecting the GUI behavior.

Two proposed solutions:
a) Add an exception like the sprint/sneak modifier. "Use normal mouse controls in menus"
b) Split mouse rebinding into action rebinding. "Break key" and "Place key" which are independent of the menus

**Comment by Kestrelis** [comment]
Adding to this issue since I don't think I warrants a separate one--let me know if this should be moved, I'm new. 
The "normal mouse controls in menus" feature would also be good for those who bind a mouse button to a keyboard button instead (me). As a result of this binding, I can't actually type that key in the chat or to name a waypoint, unless I place my cursor at the end of the textbox. This seems pretty unintended.

If the controls section allowed for users to bind break and place, I assume that would also allow various bindings for mouse-buttons (mb3, mb4...) to actions which also seems to be a requested feature. It might be nice to have both proposed solutions, though, for users who need it.

**Comment by YoshiWoof22** [comment]
Very new player here,

Spent quite some time trying to figure out why swapping the Mouse keys was also swapping inventory handling - I initially assumed it was a bug, because the behaviour very much *feels* wrong/unintended, and that is how I came across this issue.

As someone who very much prefers switching their left/right click for placing/breaking (even in other games), finding out that its also tied to the GUI/Inventory in VS is sadly quite an accessibility bummer to me.

Really hoping to still see this getting addressed at some point, which is why I'm leaving the comment here to maybe get some conversation going again about this topic.

**Comment by meganukebmp** [comment]
> Very new player here,
> 
> Spent quite some time trying to figure out why swapping the Mouse keys was also swapping inventory handling - I initially assumed it was a bug, because the behaviour certainly _feels_ wrong/unintended, and that is how I came across this issue.
> 
> As someone who very much prefers switching their left/right click for placing/breaking (even in other games), finding out that its also tied to the GUI/Inventory in VS is sadly quite an accessibility bummer to me.
> 
> Really hoping to still see this getting addressed at some point, which is why I'm leaving the comment here to maybe get some conversation going again about this topic.

Hey, I actually made a mod for myself to resolve this issue. It may be useful to you. It's been working since 1.19. All it does is flip the buttons in GUIs.

[vsmouseguiswapfix_1.0.0.zip](https://github.com/user-attachments/files/19770047/vsmouseguiswapfix_1.0.0.zip)

**Comment by YoshiWoof22** [comment]
> Hey, I actually made a mod for myself to resolve this issue. It may be useful to you. It's been working since 1.19. All it does is flip the buttons in GUIs.
> 
> [vsmouseguiswapfix_1.0.0.zip](https://github.com/user-attachments/files/19770047/vsmouseguiswapfix_1.0.0.zip)

During my search I did actually see an older mod on the VS Mod Browser that supposedly did the same thing on older versions of VS, but that one was broken.

This is absolutely amazing - thank you so much!

**Comment by meganukebmp** [comment]
> > Hey, I actually made a mod for myself to resolve this issue. It may be useful to you. It's been working since 1.19. All it does is flip the buttons in GUIs.
> > [vsmouseguiswapfix_1.0.0.zip](https://github.com/user-attachments/files/19770047/vsmouseguiswapfix_1.0.0.zip)
> 
> During my search I did actually see an older mod on the VS Mod Browser that supposedly did the same thing on older versions of VS, but that one was broken.
> 
> This is absolutely amazing and works like a charm - thank you so much!

Glad it works for you :)

I'm also the maker of the old mod on the moddb. Back then it was impossible swap the buttons at all. So that mod did that. Now that it's possible to swap the buttons it's been deprecated (and broken). I'm waiting on this issue here to be resolved rather than publishing this mod, since it's just very specific.
